**Linux** [Build 9711](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=95614&buildTypeId=WFPR) outcome was **FAILURE** using a merge of d6f3afe1321a80496cf3a6907a7ca5fabb217892
Summary: Tests failed: 1 (1 new), passed: 3318, ignored: 194 Build time: 00:56:39
### Failed tests
```
org.jboss.as.test.multinode.ejb.timer.database.DatabaseTimerServiceMultiNodeTestCase: 	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:83)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
Caused by: java.lang.Exception: {"WFLYCTL0180: Services with missing/unavailable dependencies" => ["jboss.deployment.unit.\"testTimerServiceSimple.war\".component.TimedObjectTimerServiceBean.ejb3.timerService is missing [jboss.ejb3.timerService.timerPersistence.dbstore]"]}
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getActionResult(ServerDeploymentPlanResultFuture.java:134)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.getResultFromNode(ServerDeploymentPlanResultFuture.java:123)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:85)
	at org.jboss.as.controller.client.helpers.standalone.impl.ServerDeploymentPlanResultFuture.get(ServerDeploymentPlanResultFuture.java:42)
	at org.jboss.as.controller.client.helpers.standalone.ServerDeploymentHelper.deploy(ServerDeploymentHelper.java:55)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:77)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deployInternal(ArchiveDeployer.java:64)
	at org.jboss.as.arquillian.container.ArchiveDeployer.deploy(ArchiveDeployer.java:46)
	at org.jboss.as.arquillian.container.CommonDeployableContainer.deploy(CommonDeployableContainer.java:144)
------- Stdout: -------
 [0m14:35:05,655 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 29) WFLYUT0022: Unregistered web context: /testTimerServiceSimple
 [0m [0m14:35:05,670 INFO  [org.jboss.weld.deployer] (MSC service thread 1-1) WFLYWELD0010: Stopping weld service for deployment testTimerServiceSimple.war
 [0m [31m14:35:05,672 ERROR [stderr] (Timer-1) Exception in thread "Timer-1" java.lang.NullPointerException
 [0m [31m14:35:05,672 ERROR [stderr] (Timer-1) 	at java.util.HashSet.<init>(HashSet.java:118)
 [0m [31m14:35:05,672 ERROR [stderr] (Timer-1) 	at org.jboss.as.ejb3.timerservice.persistence.database.DatabaseTimerPersistence$RefreshTask.run(DatabaseTimerPersistence.java:772)
 [0m [31m14:35:05,673 ERROR [stderr] (Timer-1) 	at java.util.TimerThread.mainLoop(Timer.java:555)
 [0m [31m14:35:05,673 ERROR [stderr] (Timer-1) 	at java.util.TimerThread.run(Timer.java:505)
 [0m [31m14:35:05,682 ERROR [org.jboss.as.ejb3] (Timer-1) WFLYEJB0456: Failed to refresh timers for testTimerServiceSimple.testTimerServiceSimple.TimedObjectTimerServiceBean
 [0m [0m14:35:05,712 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 24) WFLYCLINF0003: Stopped client-mappings cache from ejb container
 [0m [0m14:35:05,733 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0028: Stopped deployment testTimerServiceSimple.war (runtime-name: testTimerServiceSimple.war) in 87ms
 [0m [0m14:35:05,745 INFO  [org.jboss.as.repository] (management-handler-thread - 1) WFLYDR0002: Content removed from location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/multinode/target/jbossas-multinode-server/standalone/data/content/20/9130cdd384efa99accaa4fc0e2bab9ffa28a93/content
 [0m [0m14:35:05,745 INFO  [org.jboss.as.server] (management-handler-thread - 1) WFLYSRV0009: Undeployed "testTimerServiceSimple.war" (runtime-name: "testTimerServiceSimple.war")
 [0m [31m14:35:05,779 ERROR [org.jboss.as.ejb3] (Timer-1) WFLYEJB0456: Failed to refresh timers for testTimerServiceSimple.testTimerServiceSimple.TimedObjectTimerServiceBean
 [0m14:35:05,803 ERROR [org.jboss.as.arquillian.container.ServerSetupObserver] (main) Setup task failed during tear down. Offending class 'org.jboss.as.test.multinode.ejb.timer.database.DatabaseTimerServiceMultiNodeExecutionDisabledTestCase$DatabaseTimerServiceTestCaseServerSetup@4fc774': MgmtOperationException{operation={
    "operation" => "remove",
    "address" => [
        ("subsystem" => "ejb3"),
        ("service" => "timer-service"),
        ("database-data-store" => "dbstore")
    ],
    "operation-headers" => {
        "rollback-on-runtime-failure" => false,
        "allow-resource-service-restart" => true
    }
}, result={
    "outcome" => "failed",
    "failure-description" => "WFLYCTL0216: Management resource '[
    (\"subsystem\" => \"ejb3\"),
    (\"service\" => \"timer-service\"),
    (\"database-data-store\" => \"dbstore\")
]' not found",
    "rolled-back" => true,
    "response-headers" => {"process-state" => "reload-required"}
}}
	at org.jboss.as.test.integration.management.ManagementOperations.executeOperation(ManagementOperations.java:67)
	at org.jboss.as.test.integration.management.ManagementOperations.executeOperation(ManagementOperations.java:37)


```

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/CdiJpaInjectingBean.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/CdiJpaInjectingBean.java
@@ -1,0 +1,44 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.integration.weld.jpa.subdeployment;
+
+import javax.enterprise.inject.Produces;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.persistence.PersistenceContextType;
+import javax.persistence.PersistenceUnit;
+import javax.persistence.Query;
+
+public class CdiJpaInjectingBean {
+
+    @Produces
+       @QualifyEntityManagerFactory
+       @PersistenceUnit(unitName="cdiPu")
+    EntityManagerFactory emf;
+
+    public EntityManagerFactory entityManagerFactory() {
+        return emf;
+    }
+
+
+}

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/QualifyEntityManagerFactory.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/QualifyEntityManagerFactory.java
@@ -1,0 +1,46 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.integration.weld.jpa.subdeployment;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.inject.Qualifier;
+
+/**
+ * QualifyEntityManagerFactory
+ *
+ * @author Scott Marlow
+ */
+@Qualifier
+@Retention(RUNTIME)
+@Target({TYPE, METHOD, FIELD, PARAMETER})
+public @interface QualifyEntityManagerFactory {
+}

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/WeldSubdeploymentScopeTestCase.java
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/WeldSubdeploymentScopeTestCase.java
@@ -1,0 +1,72 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.integration.weld.jpa.subdeployment;
+
+import javax.inject.Inject;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.EnterpriseArchive;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * WFLY-6485
+ * <p/>
+ * Test that JPA dependencies are set for sub-deployments
+ *
+ * @author Scott Marlow
+ */
+@RunWith(Arquillian.class)
+public class WeldSubdeploymentScopeTestCase {
+
+    @Deployment
+    public static Archive<?> deploy() {
+        EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "cdiPuScope.ear");
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "simple.war");
+        war.addClasses(WeldSubdeploymentScopeTestCase.class, CdiJpaInjectingBean.class);
+        war.addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        JavaArchive jar = ShrinkWrap.create(JavaArchive.class, "util.jar");
+        jar.addAsManifestResource(WeldSubdeploymentScopeTestCase.class.getPackage(), "persistence.xml", "persistence.xml");
+        war.addAsLibrary(jar);
+        ear.addAsModule(war);
+        return ear;
+    }
+
+    @Inject
+    private CdiJpaInjectingBean bean;
+
+    @Test
+    public void testInjectedPersistenceContext() throws Exception {
+        Assert.assertNotNull("WFLY-6485 regression, injected EntityManagerFactory should not be null but is",
+                bean.entityManagerFactory());
+    }
+}
+

--- a/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/persistence.xml
+++ b/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/weld/jpa/subdeployment/persistence.xml
@@ -1,0 +1,32 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2016, Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags. See the copyright.txt file in the
+  ~ distribution for a full listing of individual contributors.
+  ~
+  ~ This is free software; you can redistribute it and/or modify it
+  ~ under the terms of the GNU Lesser General Public License as
+  ~ published by the Free Software Foundation; either version 2.1 of
+  ~ the License, or (at your option) any later version.
+  ~
+  ~ This software is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<persistence xmlns="http://java.sun.com/xml/ns/persistence"	version="2.0">
+	<persistence-unit name="cdiPu">
+		<description>Persistence Unit.</description>
+		<jta-data-source>java:jboss/datasources/ExampleDS</jta-data-source>
+		<properties>
+			<property name="hibernate.hbm2ddl.auto" value="create-drop" />
+		</properties>
+	</persistence-unit>
+</persistence>

--- a/weld/src/main/java/org/jboss/as/weld/deployment/processors/WeldDeploymentProcessor.java
+++ b/weld/src/main/java/org/jboss/as/weld/deployment/processors/WeldDeploymentProcessor.java
@@ -180,7 +180,7 @@ public class WeldDeploymentProcessor implements DeploymentUnitProcessor {
         getJpaDependencies(deploymentUnit, jpaServices);
 
         for (DeploymentUnit subDeployment : subDeployments) {
-            getJpaDependencies(deploymentUnit, jpaServices);
+            getJpaDependencies(subDeployment, jpaServices);
             final Module subDeploymentModule = subDeployment.getAttachment(Attachments.MODULE);
             if (subDeploymentModule == null) {
                 continue;

